CPAC/utils/serialization/core.py

```diff
@@ -0,0 +1,39 @@
+from typing import Any
+
+from nipype import __version__ as version_nipype
+from nipype.pipeline.engine import Workflow, Node
+
+from CPAC import __version__ as version_cpac
+
+WorkflowRaw = Workflow
+"""Alias for internal use. CPAC derives: CPAC.pipeline.nipype_pipeline_engine.engine.Workflow"""
+
+NodeRaw = Node
+"""Alias for internal use. CPAC derives: CPAC.pipeline.nipype_pipeline_engine.engine.Node"""
+
+VERSION_WORKFLOW = 1
+"""Placeholder for version checking. Increase when file structure changes drastically."""
+
+
+def workflow_container(workflow: Any, meta: Any = None) -> dict:
+    """
+    Construct a container dictionary with some version information to allow version checks when reading.
+
+    Parameters
+    ----------
+    workflow : Workflow object.
+    meta : Meta information.
+
+    Returns
+    -------
+
+    """
+    return {
+        'version': {
+            'workflow': VERSION_WORKFLOW,
+            'cpac': version_cpac,
+            'nipype': version_nipype
+        },
```

Comment:
I'm not opposed to this verbosity, but I wonder if just the C-PAC version would suffice since Nipype is packaged in the container and `VERSION_WORKFLOW` is hardcoded in this file

Reply:
I like it! I just think

```Python
'version': version_cpac
```

might cover everything instead of needing the whole sub-dictionary

Reply:
(The subdict is fine though, no need to remove it if you prefer the verbosity)

Reply:
In the intrest of future-proofing in case we go ahead with moving it in a seperate package I sggest 
```python
'version': {
    'cpac': version_cpac,
},
```
this way we can easily add the package version without having to change the viewer code.
The main thing I am worried about is that the CPAC version strings are not super easy to parse (major, minor, patch + development versions etc.) vs just having an integer to `>` against.
But it's your call 👍👎

Reply:
I'm fine to keep it as is, and we can revisit later if necessary!